### PR TITLE
Add auth prop to support urql client

### PR DESCRIPTION
**Pull request by theianjones**
[Heres an example app](https://github.com/theianjones/urql-graphql/tree/master/src) of why i need this change.

### CODE REVIEW

File: packages/react-onegraph/src/index.js [hunk 1]
```diff
@@ -1,90 +1,97 @@
...
+    )
+  }
+
+  getAuth = () => {
+    const auth =
+      this.props.auth ||
+      new OneGraphAuth({
+        appId: this.props.appId,
+      })
+    return auth
```
Comment (theianjones) [inline packages/react-onegraph/src/index.js:34]: Use auth if its passed in, else default to creating an auth ourselves.

File: packages/react-onegraph/src/index.js [hunk 2]
```diff
@@ -1,90 +1,97 @@
...
+          out[service] = status[service].isLoggedIn
+          return out
         }, {}),
         auth,
       }),
-    );
+    )
+  }
+
+  getAuth = () => {
```
Comment (dwwoelfel) [inline packages/react-onegraph/src/index.js:28]: Can you move this logic into the state initialization. I think you should be able to do:

```js
state = {
  auth: this.props.auth || new OneGraphAuth({appId: this.props.appId})
 ...
}
```

If that doesn't work, then putting in the constructor would also be fine (it's not side-effecting, so it doesn't need to go in `componentDidMount`).

File: packages/react-onegraph/src/index.js [hunk 3]
```diff
@@ -1,90 +1,97 @@
-import React, {Component, createContext} from 'react';
-import OneGraphAuth from 'onegraph-auth';
+import React, {Component, createContext} from 'react'
```
Comment (dwwoelfel) [inline packages/react-onegraph/src/index.js:1]: Can you run `yarn prettier`. That should remove the formatting changes in the diff.

**Comment by theianjones** [comment]
@dwwoelfel got those changes in!

**Review by dwwoelfel — APPROVED** [review]
Oops, I let this slip through! Merging now...

**Comment by dwwoelfel** [comment]
npm seems to be down--will publish as 3.0.x when it's back up

**Comment by dwwoelfel** [comment]
Published as react-onegraph@3.0.2

**Pull request merged**